### ISSUE TITLE
false positive `unused-qualifications`

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust#114433]
### Code

```Rust
#![warn(unused_qualifications)]

fn bar3() {
    match Option::<Option<()>>::None {

/*
help: remove the unnecessary path segments
  |
4 -     match Option::<Option<()>>::None {
4 +     match None {
  |
*/
        Some(v) => {}
        None => {}
    }
}

fn main() {}
```

### Current output

```Shell
warning: unnecessary qualification
 --> q.rs:4:11
  |
4 |     match Option::<Option<()>>::None {
  |           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
note: the lint level is defined here
 --> q.rs:1:9
  |
1 | #![warn(unused_qualifications)]
  |         ^^^^^^^^^^^^^^^^^^^^^
help: remove the unnecessary path segments
  |
4 -     match Option::<Option<()>>::None {
4 +     match None {
  |
```

### Rationale and extra context

maybe don't lint? the suggested `None` breaks type inference
